crates/rpc/rpc-eth-api/src/helpers/transaction.rs

```diff
@@ -555,17 +391,20 @@ pub trait EthTransactions: LoadTransaction {
     fn sign_request(
         &self,
         from: &Address,
-        request: TypedTransactionRequest,
-    ) -> Result<TransactionSigned, Self::Error> {
-        for signer in self.signers().read().iter() {
-            if signer.is_signer_for(from) {
-                return match signer.sign_transaction(request, from) {
-                    Ok(tx) => Ok(tx),
-                    Err(e) => Err(e.into_eth_err()),
+        txn: TransactionRequest,
+    ) -> impl Future<Output = Result<TransactionSigned, Self::Error>> + Send {
+        async move {
+            let signers: Vec<_> = self.signers().read().iter().cloned().collect();
+            for signer in signers {
+                if signer.is_signer_for(from) {
+                    return match signer.sign_transaction(txn.clone(), from).await {
```

Comment:
I don't think we need the clone here
```suggestion
                    return match signer.sign_transaction(txn, from).await {
```